js/ui/shellEntry.js

```diff
@@ -184,6 +184,26 @@ var OverviewEntry = GObject.registerClass({
         global.stage.connect('notify::key-focus', this._onStageKeyFocusChanged.bind(this));
     }
 
+    _updateSearchEntryText() {
+        let entryText;
+        let internetProvider = InternetSearch.getInternetSearchProvider();
```

Comment:
Was checking this code and looks like the call to `InternetSearch.getInternetSearchProvider()` will always return a new object, wonder if that could be an issue or waste of resources (specially considering https://phabricator.endlessm.com/T27082). I wonder if we could access the Search.SearchResults here and check whether the cached internet provider (after reloading it when the parental controls settings change) is valid?

Reply:
the `ViewsDisplay` class from (`viewSelector.js`) seems to have access to both the `SearchResults` and the `OverviewEntry` objects so it may be a good place to add the plumbing.

Reply:
Isn't this getting a bit complicated for read / maintain the code? 
I mean,  I get your point, but this is something that might hit back during next rebase etc. 
It's exposing elements (e.g. shell entry hint text, cached providers) to other classes, because to read/write their values based on relevant `if .. else` cases.
Meanwhile I think, the `ViewsDisplay` class shouldn't be getting touched here at all.

Reply:
Yes, I agree this may be getting a bit complicated for the use case, but at the same time I am not sure about the extra resources used by this impl. Maybe we should check what others think? @pwithnall and/or @wjt what do you think? The changes look good, but see comments above.

Reply:
I don't think there is a performance problem here. This function will only be called when the entry is constructed (once) and when parental controls change (rarely), and the returned object will only live until the next GC pass.

Reply:
Ok, fair enough, lets go with it then. This looks good apart from the small comments below.